cfpb/cfgov-refresh
# Better error handling/testing for external redirect page

This commit makes some slight changes to the `ExternalURLForm` and `ExternalURLNoticeView` that uses that form. It now properly handles invalid `POST`s (previously this would cause a server error), fixes a bug with validating URLs with unicode characters, and adds some additional test coverage.

## Additions

- New unit tests for `ExternalURLNoticeView`.

## Testing

New unit tests should cover all cases, but, to verify:

1. Locally visit the homepage at [http://localhost:8000](http://localhost:8000).
2. Click on the Facebook link in the footer (http://localhost:8000/external-site/?ext_url=https%3A%2F%2Ffacebook.com%2Fcfpb); note this goes to an interstitial page.
3. Mess with the query string of that page (e.g. change the URL to an non-whitelisted site, like http://localhost:8000/external-site/?ext_url=https%3A%2F%2Ffacebook.com%2Fnotcfpb) and note that you get a 404.
4. Visit an invalid URL with a unicode character (e.g. http://localhost:8000/external-site/?ext_url=https://www.facebook.com/CFPB-La-Oficina-para-la-Protecci%C3%B3n-Financiera-del-Consumidor) and note that you get a 404 instead of a 500.
5. Visit a valid URL with a unicode character (requires creating e.g. a Browse Page, adding a rich text field, adding an external link to a URL [like this](https://ko.wikipedia.org/wiki/%EC%9C%84%ED%82%A4%EB%B0%B1%EA%B3%BC:%EB%8C%80%EB%AC%B8), and previewing the page. Verify that the interstitial page works properly.

## Checklist
